src/router/response.rs

```diff
@@ -10,35 +11,65 @@ use tokio_io;
 
 static NOTFOUND: &[u8] = b"Not Found";
 
+pub fn body(body: impl ResponseBody) -> Result<Response<Body>, Error> {
+    let body = body.into_body();
+    Response::builder().status(StatusCode::OK).body(body)
+}
+
+pub fn json(body: impl Serialize) -> Result<Response<Body>, Error> {
+    let serialized_obj = match serde_json::to_string(&body) {
+        Ok(val) => val,
+        Err(e) => std::error::Error::description(&e).to_string(),
+    };
+
+    let body = serialized_obj.into_body();
+
+    Response::builder().status(StatusCode::OK).body(body)
+}
+
+pub fn json_with_status(
```

Comment:
I think builder style is better.

Reply:
for example?

Reply:
Since this function is used internally, we can use something like ```json_serialize``` to return the serialize json string and do the Response::builder in ```impl Responder for ResponseType<T>```. 
This design is more open and extensible I think as developer may want to do their own implementation for ```Responder for OwnJsonType```. In certain case, they might